scripts/pre_commit_linter.py

```diff
@@ -580,6 +580,94 @@ def _check_bad_pattern_in_file(filename, content, pattern):
     return False
 
 
+def _check_bad_pattern_in_comments_docstrings(filename, content):
```

Comment:
Since we are moving towards custom lint checks, it would be better if we did something like [this](https://stackoverflow.com/questions/8724851/tool-for-automatically-check-docstring-style-according-to-pep257). This looks much cleaner and is more efficient than functions using regex.

Reply:
@seanlip: Can you please have a look at this point?

Reply:
@apb7 just to clarify, what exactly are you suggesting? Import the third-party lib?

In general I would say that anything that requires less ongoing maintenance on our part is better.

Reply:
@seanlip: It would be better if we write a custom Pylint checker for this, something like:
```
class DocStringChecker(BaseChecker):
    """
    PyLint AST based checker to eval compliance with PEP 257-ish conventions.
    """
    __implements__ = IASTNGChecker

    name = 'doc_string_checker'
    priority = -1
    msgs = {'W9001': ('One line doc string on >1 lines',
                     ('Used when a short doc string is on multiple lines')),
            'W9002': ('Doc string does not end with "." period',
                     ('Used when a doc string does not end with a period')),
            'W9003': ('Not all args mentioned in doc string',
                     ('Used when not all arguments are in the doc string ')),
            'W9004': ('triple quotes',
                     ('Used when doc string does not use """')),
           }
    options = ()

    def visit_function(self, node):
        if node.doc: self._check_doc_string(node)
```
We can modify this class according to our need.  
Thanks!

Reply:
Yes, I'd agree with that. I think it makes more sense to build it into the existing linter system.

Reply:
Hi @apb7, the check which we are currently doing should not be added as a custom check to pylint since it checks for most of the cases but still there are cases which cannot be detected. It is not possible for any checker to detect where a period should be present in case of multiline comments.
Also, the approach defined in the stackoverflow link results in a lot of errors. I searched for alternate approaches but I was unable to find any suitable tutorial on it since the problem remains the same, we cannot check for all comments. In case of multiline comments, we need to go to the last line for the check since it is not feasible to check every line.